deps/require.lua

```diff
@@ -29,14 +29,14 @@ local luvi = require('luvi')
 local bundle = luvi.bundle
 local pathJoin = luvi.path.join
 local env = require('env')
-local os = require('ffi').os
 local uv = require('uv')
 
 local realRequire = _G.require
 
-local tmpBase = os == "Windows" and (env.get("TMP") or uv.cwd()) or
-                                    (env.get("TMPDIR") or '/tmp')
-local binExt = os == "Windows" and ".dll" or ".so"
+local tmpBase = env.get("TMPDIR") or env.get("TMP") or env.get("TEMP") or (uv.fs_access("/tmp", "r") and "/tmp") or uv.cwd()
+
+local first_cpath = package.cpath:match("[^" .. package.config:sub(3, 3) .. "]+")
+local binExt = (first_cpath and first_cpath:match("%.[^.]+$")) or (package.config:sub(1, 1) == "\\" and ".dll" or ".so")
```

Comment:
What is the reason for this `binExt` change?

Reply:
Require has no dependencies on luvit libraries, so it can't use `los` to get the os name. The `binExt` change makes `require` take the system binary extension from the first entry in package.cpath; and only use a naive check if the system uses backslashes as a backup.

This primarily makes it trivial to support macos, which may use exclusively `.dylib` libraries.